python/plugins/processing/gui/menus.py

```diff
@@ -281,6 +283,47 @@ def getMenu(name, parent):
 
 def findAction(actions, alg):
     for action in actions:
-        if action.data() == alg.id():
+        if (isinstance(alg, str) and action.data() == alg) or (isinstance(alg, QgsProcessingAlgorithm) and action.data() == alg.id()):
             return action
     return None
+
+
+def addToolBarButton(algId, toolbar, icon=None, tooltip=None):
+    alg = QgsApplication.processingRegistry().algorithmById(algId)
+    if alg is None or alg.id() != algId:
+        QgsMessageLog.logMessage(Processing.tr('Invalid algorithm ID for button: {}').format(algId), Processing.tr('Processing'))
```

Comment:
Maybe we should be obnoxious and raise an assert here instead? It'd avoid the ongoing issue we have with the algorithm menu actions disappearing after porting them to c++...

Reply:
Done.